pkg/utils/utils.go

```diff
@@ -28,3 +28,12 @@ func ManagedCredentialsProvider(e *operatorv1alpha1.ExternalDNS) bool {
 	}
 	return false
 }
+
+// EnvProxySupportedProvider returns true if the ExternalDNS provider supports the proxy settings via environment variables HTTP(S)_PROXY, NO_PROXY
+func EnvProxySupportedProvider(e *operatorv1alpha1.ExternalDNS) bool {
```

Comment:
isProxyEnvSupportedByProvider() will be good, because we are using this API for decision making.

Reply:
Yeah, your name is good. I just kept the same pattern as for the other function in this file: `ManagedCredentialsProvider`.